connect/runtime/src/main/java/org/apache/kafka/connect/runtime/distributed/DistributedHerder.java

```diff
@@ -1016,7 +1016,7 @@ public void run() {
                                 String reconfigUrl = RestServer.urlJoin(leaderUrl(), "/connectors/" + connName + "/tasks");
                                 RestClient.httpRequest(reconfigUrl, "POST", taskProps, null, config);
                                 cb.onCompletion(null, null);
-                            } catch (ConnectException e) {
+                            } catch (ConnectException | NullPointerException e) {
```

Comment:
I'd rather not use exception handling for branching. I think I'd prefer to address this case with an earlier check for `leaderUrl() == null` and instantiate a more informative exception to pass to `cb.onCompletion`